.github/actions/create-jira-issues-for-community-activities/action.yml

```diff
@@ -0,0 +1,103 @@
+﻿name: Create Jira issues for community activities
+description: >
+  Creates Jira issues for community activities happening on GitHub, like issues, discussions, pull requests being
+  opened.
+
+inputs:
+  issue-component:
+    required: false
+    description: Optional component of the issues being created.
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Scripts
+      shell: pwsh
+      run: |
+        "${{ github.action_path }}" >> $Env:GITHUB_PATH
+        cd ${{ github.action_path }}
+        cd ../
+        "$((Get-Location).Path)/verify-submodule-pull-request" >> $Env:GITHUB_PATH
+
+    # Don't do anything for pull requests that are already related to an issue, i.e. their titles start with an issue
+    # key.
+    - name: Check for Jira Issue-Related Pull Request
+      id: issue-related-pr
+      shell: pwsh
+      run: |
+        Write-Output "::set-output name=is-not-issue-pr::$(-not(Check-PullRequestTitle ${{ github.event.pull_request.title }}))"
+
+    - name: Login to Jira
+      if: steps.issue-related-pr.outputs.is-not-issue-pr == 'True'
+      # v1.0.0.
+      # This will show a "Logged in as: undefined" message if the login works, see:
+      # https://github.com/atlassian/gajira-login/issues/30.
+      uses: atlassian/gajira-login@90a599561baaf8c05b080645ed73db7391c246ed
+      env:
+        JIRA_BASE_URL: ${{ env.JIRA_BASE_URL }}
+        JIRA_USER_EMAIL: ${{ env.JIRA_USER_EMAIL }}
+        JIRA_API_TOKEN: ${{ env.JIRA_API_TOKEN }}
+
+    - name: Initialize Jira Issue Details
+      if: steps.issue-related-pr.outputs.is-not-issue-pr == 'True'
+      id: issue-details
+      shell: pwsh
+      run: |
+        # The funny indentation is necessary for here-strings.
+        $github = @"
+        ${{ toJSON(github) }}
+        "@ | ConvertFrom-Json
+
+        Initialize-IssueDetails $github
+
+        $fields = @"
+        {"labels": ["created-from-github"]
+        "@
+
+        if ("${{ inputs.issue-component}}" -ne "")
+        {
+            $fields += ', "components": [{"name": "${{ inputs.issue-component }}"}]'
+        }
+
+        $fields += '}'
+
+        Write-Output "::set-output name=fields::$fields"
+
+    - name: Create Jira Issue
+      if: steps.issue-related-pr.outputs.is-not-issue-pr == 'True'
+      id: create-issue
+      uses: atlassian/gajira-create@c0a9c69ac9d6aa063fed57201e55336ada860183
+      with:
+        project: ${{ env.JIRA_PROJECT_KEY }}
+        issuetype: ${{ steps.issue-details.outputs.type }}
+        summary: ${{ steps.issue-details.outputs.summary }}
+        description: ${{ fromJSON(steps.issue-details.outputs.jsonDescription) }}
+        fields: ${{ steps.issue-details.outputs.fields }}
+
+    - name: Add Remote Link to Jira Issue
+      if: steps.issue-related-pr.outputs.is-not-issue-pr == 'True'
+      shell: bash
```

Comment:
How come this is bash?

Reply:
Curl was the easiest way to do this, and all the Jira API examples were using curl from bash. But I changed it to PS.